src/RulesContainer.js

```diff
@@ -100,20 +101,20 @@ export default class RulesContainer {
    * Register a rule in `.map` and `.classes` maps.
    */
   register(rule: Rule): void {
-    if (rule.name) this.map[rule.name] = rule
-    if (rule.className && rule.name) this.classes[rule.name] = rule.className
-    if (rule.selector) this.map[rule.selector] = rule
+    this.map[rule.key] = rule
+    if (rule instanceof RegularRule) {
+      this.map[rule.selector] = rule
+      this.classes[rule.key] = rule.selector.substr(1)
```

Comment:
this is still bad, need a better idea, selector might be any complex selector there.